tests/jenkins/TestPrintArtifactDownloadUrlsForStaging.groovy

```diff
@@ -6,23 +6,53 @@
  * compatible open source license.
  */
 
-package jenkins.tests
-
+import jenkins.tests.BuildPipelineTest
 import org.junit.*
 
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.core.IsNull.notNullValue
+
 
 class TestPrintArtifactDownloadUrlsForStaging extends BuildPipelineTest {
 
+    private List artifactFileNames
+    private String uploadPath
+
+    public LibTester libTester = new LibTester()
+
     @Before
     void setUp() {
+
+        this.registerLibTester(new TestPrintArtifactDownloadUrlsForStaging(
+                artifactFileNames: ['dummy_file.tar.gz', 'dummy_file.tar.gz.sig', 'a_dummy_file.tar.gz'],
+                uploadPath: 'dummy/upload/path'
+        ).libTester)
+
         super.setUp()
-        binding.setVariable('filenamesForUrls', ['dummy_file.tar.gz', 'dummy_file.tar.gz.sig'])
-        binding.setVariable('UPLOAD_PATH', 'dummy/upload/path')
     }
 
     @Test
     void testPrintArtifactDownloadUrlsForStaging() {
         super.testPipeline("tests/jenkins/jobs/PrintArtifactDownloadUrlsForStaging_Jenkinsfile")
     }
 
+    class LibTester extends LibFunctionTester {
```

Comment:
Make a new class file for the lib tester functionality such as `TestPrintArtifactDownloadUrlsForStagingLibTester`.  This subclass access pattern is strange, lets pivot towards well named single purposed classes.